src/app/reporting/Engine.cpp

```diff
@@ -302,9 +306,9 @@ CHIP_ERROR Engine::BuildSingleReportDataEventReports(ReportDataMessage::Builder
         else if ((err == CHIP_ERROR_BUFFER_TOO_SMALL) || (err == CHIP_ERROR_NO_MEMORY))
         {
             // when first cluster event is too big to fit in the packet, ignore that cluster event.
-            if (eventCount == 0)
+            // However, we may have encoded some attributes befure, we don't skip it in that case.
+            if (eventCount == 0 && !aBufferIsUsed)
             {
-                eventMin++;
```

Comment:
Shouldn't this still increment eventMin?  How does that match the "ignore that cluster event" comments above?

Reply:
Yes, this is removed by mistake, eventMin should be increased by 1 when the buffer is clean before encoding.